src/main/java/ca/on/oicr/pineryreports/reports/impl/EmptyStockReport.java

```
@@ -0,0 +1,170 @@
+        filters.add(byCreator(userIds));
+        filters.add(hasDescendants(allSamplesById));
+        filters.add(byCreatedBetween(start, end));
+        filters.add(sample -> sample.getVolume() != null && sample.getVolume() <= 0);
+        return filter(unfiltered, filters);
+    }
+
+    /** Sort created date, then name */
+    private final Comparator<SampleDto> byReceiveDateAndName =
+            (dto1, dto2) -> {
+                String dto1Created = removeTime(dto1.getCreatedDate());
+                String dto2Created = removeTime(dto2.getCreatedDate());
+                int byDate = 0;
+                if (dto1Created == null) {
+                    if (dto2Created != null) {
+                        byDate = 1;
+                    }
+                } else if (dto2Created == null) {
+                    byDate = -1;
+                } else {
+                    byDate = dto1Created.compareTo(dto2Created);
+                }
+                return byDate == 0 ? dto1.getName().compareTo(dto2.getName()) : byDate;
+            };
+
+    @Override
+    protected List<ColumnDefinition> getColumns() {
+        return COLUMNS;
+    }
+
+    @Override
+    protected int getRowCount() {
+        return stocks.size();
+    }
+
+    @Override
+    protected String[] getRow(int rowNum) {
+        SampleDto stock = stocks.get(rowNum);
+        String[] row = new String[COLUMNS.size()];
+
+        row[0] = stock.getId();
+        row[1] = stock.getName();
+        row[2] = removeTime(stock.getCreatedDate());
+        row[3] = removeTime(stock.getModifiedDate());
+        row[4] = toStringOrNull(stock.getModifiedById());
+        row[5] = getUpstreamAttribute(ATTR_EXTERNAL_NAME, stock, allSamplesById);
+        Float concentration = stock.getConcentration();
+        row[6] = concentration == null ? null : round(concentration, 2);
+        Float initVolume = getFloatAttribute(ATTR_INITIAL_VOLUME, stock, 0F);
```

Defaulting to zero might be misleading if it wasn't set